Team2/fix onboarding

Quinxie51 opened the pull request:
## 🚀 What does this PR do?

> Refactor the onboarding flow by deleting `Name.tsx`, `Birthday.tsx`, and `Photo.tsx`, and consolidating them into a single `OnboardingFlow.tsx` component.  

> This makes the user onboarding experience simpler and more maintainable by handling user input (name, birthdate, photo) with centralized React state.

---

## ✅ Checklist

- [x] Code compiles & runs
- [x] Lint checks pass
- [x] Deleted `Name.tsx`, `Birthday.tsx`, and `Photo.tsx`
- [x] Created `OnboardingFlow.tsx` with centralized state management
- [x] Added `printInfo()` to log user data at the end
- [ ] Updated `.env.example` (N/A for this PR)
- [x] PR linked to related issue/ticket (if applicable)

---

## 🔍 How to test this

1. Run `npx expo start`
2. Navigate to the onboarding screen
3. Complete the steps:
   - Enter a full name
   - Select a birthdate
   - Upload a profile photo
4. On the final step, verify that all information (name, birthdate, photo URI) is logged correctly in the console.
5. Ensure the app transitions smoothly between steps without routing issues.
6. Check that no crashes occur on both Android and iOS simulators/devices.

---

## 🧠 Extra Notes

- `expo-router` is still used for internal navigation handling where necessary.
- For now, the "Finish" button redirects to a placeholder page `/page` — needs update once the actual home/dashboard is ready.
- No changes needed for environment variables (`.env`) for this PR.
- Future improvement: validate user input (e.g., prevent empty name submission).

Quinxie51 committed:
merge 3 files into onboardingFlow

Quinxie51 committed:
delete old files

yourvivian commented on app/(onboard)/onboardingFlow.tsx:
```diff
@@ -0,0 +1,187 @@
+  };
+
+  const handleNext = () => setStep((prev) => prev + 1);
+  const handleBack = () => setStep((prev) => prev - 1);
```
Unused function

ThanhDatVu111 replied:
we use step state now instead of those functions

yourvivian commented on app/(onboard)/onboardingFlow.tsx:
```diff
@@ -0,0 +1,187 @@
+          className="bg-accent py-3 rounded-lg w-full items-center mt-6"
+          onPress={() => {
+            router.push('/birthday');
+            setStep((prevStep: number) => prevStep + 1);
```
Unnecessary route change?

ThanhDatVu111 replied:
we not use router push any more , since name , birthday, profile photo into one file and we switch between those functions using step state

yourvivian commented on app/(onboard)/onboardingFlow.tsx:
```diff
@@ -0,0 +1,187 @@
+import DateTimePicker from '@react-native-community/datetimepicker';
+import * as ImagePicker from 'expo-image-picker';
+
+import { useAuth, useUser } from '@clerk/clerk-expo';
```
remove unusued import

ThanhDatVu111 replied:
this is done

yourvivian commented on app/(onboard)/onboardingFlow.tsx:
```diff
@@ -0,0 +1,187 @@
+    console.log("name:" + name);
+    console.log("birthday:" + birthdate);
+  }
+
```
remove unused func

ThanhDatVu111 replied:
this is done

yourvivian commented on app/(onboard)/onboardingFlow.tsx:
```diff
@@ -0,0 +1,187 @@
+          borderColor="#F5829B"
+          value={name}
+          placeholder="Kevin Nguyen"
+          onChangeText={(text: string) => setFullName(text)}
```
Do we want another placeholder?

ThanhDatVu111 replied:
like another name ?

yourvivian replied:
Just say "Your name"

yourvivian commented on app/(auth)/signup.tsx:
Use Button component instead of repeating TouchableOpacity styling

ThanhDatVu111 replied:
yes this is done

yourvivian commented on app/(auth)/signup.tsx:
```diff
@@ -49,7 +47,13 @@ export default function SignUpScreen() {
+          });
       } else {
         console.error(JSON.stringify(signUpAttempt, null, 2));
       }
```
Do we want to handle error with readableMessage like we did with `onSignUpPress`

ThanhDatVu111 replied:
this is done

yourvivian commented on app/(onboard)/onboardingFlow.tsx:
```diff
@@ -0,0 +1,187 @@
+  const [step, setStep] = useState(0);
+  const [name, setFullName] = useState('');
+  const [photo, setPhoto] = useState<string | null>(null);
+  const [birthdate, setBirthdate] = useState(new Date());
```
```suggestion
  const [userData, setUserData] = useState({
    name: '',
    birthdate: new Date(),
    photo: null,
  });
```

then a function like
```
const updateUserData = (key: string, value: any) => {
  setUserData((prev) => ({ ...prev, [key]: value }));
};
```

and use like   
```
onChangeText={(text: string) => updateUserData('name', text)}
```

or use `useReducer`

ThanhDatVu111 committed:
Refactor onboarding flow and button for customization

- Update SignUp screen to pass correct userId (from Clerk’s useAuth) and email into onboarding params
- Refactor OnboardingFlow to simplify step components (NameStep, BirthdayStep, PhotoStep) for clarity
- Replace inline touchables with the Button component throughout steps
- Simplify Button component by removing default size and color props, allowing full styling customization via props

MC2114 committed:
Trivial commit

MC2114 committed:
Co-authored-by: Thanh Dat Vu <ThanhDatVu111@users.noreply.github.com>
Co-authored-by: quinxie <Quinxie51@users.noreply.github.com>
Co-authored-by: aanh1009 <aanh1009@users.noreply.github.com>

MC2114 committed:
Insert new user data onto a new row on Supabase remote DB

Quinxie51 committed:
change the button to button component

Quinxie51 committed:
add previous steps to on boarding

ThanhDatVu111 committed:
just some  minor changes and check quyen commit

ThanhDatVu111 commented on app/(auth)/signin.tsx:
```diff
@@ -72,54 +73,49 @@ export default function Page() {
+  textClassName="text-white text-[16px]"
+  textStyle={{ fontFamily: 'Poppins-Regular' }}
+/>
 
```
Good work but in sign up file still have button that created using </TouchableOpacity>, please change it to button component and make sure to check everyfile again to see if anything use  </TouchableOpacity> to create the button

Quinxie51 committed:
switch into button component in sign up page

Quinxie51 committed:
Co-authored-by: Chau Hoang Minh Tran <MC2114@users.noreply.github.com>
Co-authored-by: Thanh Dat Vu <ThanhDatVu111@users.noreply.github.com>
Co-authored-by: aanh1009 <aanh1009@users.noreply.github.com>

ThanhDatVu111 committed:
Add error handling for verification and implement resend code feature with countdown timer; clean up React imports

MC2114 reviewed (approved):
Code looks clean and well structured. When I run it on my machine, the UI looks good, the code flows perfectly.
